File: tests/makecode/context-switch-time/context-switch-test.ts
function set_gpio(state: number) {
    pins.digitalWritePin(DigitalPin.P1, state)
}

function high() {
    while (1) {
        set_gpio(1);
        control.spinScheduler();
    }
}

function low() {
    while (1) {
        set_gpio(0);
        control.spinScheduler();
    }
}

set_gpio(0);

control.inBackground(high)
control.inBackground(low)

control.releaseFiber();

File: tests/codal/context-switch-time/ContextSwitchTest.cpp
#include "MicroBit.h"

MicroBit uBit;

#define BUFFER_SIZE 0

void set_gpio(int pin_number, int state)
{
    switch(pin_number)
    {
        case 0:
            uBit.io.P0.setDigitalValue(state);
            break;
        case 1:
            uBit.io.P1.setDigitalValue(state);
            break;
        case 2:
            uBit.io.P2.setDigitalValue(state);
            break;
    }
}

void high()
{
#if BUFFER_SIZE > 0
    volatile char b[BUFFER_SIZE] = {0};
#endif


    while(1)
    {
        set_gpio(1,1);
        schedule();
    }
}

void low()
{
#if BUFFER_SIZE > 0
    volatile char b[BUFFER_SIZE] = {0};
#endif

    while(1)
    {
        set_gpio(1,0);
        schedule();
    }
}


int main()
{
    uBit.init();

    set_gpio(1,0);

    create_fiber(low);
    create_fiber(high);

    release_fiber();
}
